### PR TITLE
[cli] Document adding telemetry tracking

### PR DESCRIPTION
**Pull request by trek**
Adds docs for telemetry and guidance on adding new ones.

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ import { ProxyAgent } from 'proxy-agent';
 import box from './util/output/box';
 import { execExtension } from './util/extension/exec';
 import { TelemetryEventStore } from './util/telemetry';
-import { TelemetryBaseClient } from './util/telemetry/base';
```
Comment (trek) [inline packages/cli/src/index.ts:56]: `Base` really made this sound like an abstract base class when it's not. Trying on `root` for size. Also changed the naming pattern to match the others of `{CommandName}TelemetryClient`

**Review by EndangeredMassa — DISMISSED** [review]

File: packages/cli/src/index.ts (hunk 1 above)
Comment (jeffsee55) [reply packages/cli/src/index.ts:56]: Agree!

**Review by jeffsee55 — DISMISSED** [review]

**Review by jeffsee55 — DISMISSED** [review]

**Review by onsclom — DISMISSED** [review]
Very helpful README!

**Review by erikareads — APPROVED** [review]

File: packages/cli/test/unit/index.test.ts
Comment (erikareads) [inline packages/cli/test/unit/index.test.ts:1]: there are a few more calls to `TelemetryBaseClient` that need to be fixed

**Review by erikareads — DISMISSED** [review]

File: packages/cli/test/unit/index.test.ts
Comment (erikareads) [reply packages/cli/test/unit/index.test.ts:1]: https://github.com/vercel/vercel/pull/12208/files#diff-852d992d60b76ad80cf6b9ddcb0d5243c4093b33422f586d8cd129a2e51c9084R22

Comment (erikareads) [reply packages/cli/test/unit/index.test.ts:1]: https://github.com/vercel/vercel/pull/12208/files#diff-852d992d60b76ad80cf6b9ddcb0d5243c4093b33422f586d8cd129a2e51c9084R44

**Review by erikareads — APPROVED** [review]

**Pull request merged**